### PR TITLE
fix SinkSpec, and async boundary attribute, #22463

### PR DESCRIPTION
**Pull request by patriknw**
* let's keep the AsyncBoundary attribute for now,
  makeIsland is done in setAttributes
* fixing SinkSpec and RunnableGraphSpec

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/TraversalBuilder.scala [hunk 1]
```diff
@@ -267,11 +267,8 @@ sealed trait TraversalBuilder {
   protected def internalSetAttributes(attributes: Attributes): TraversalBuilder
 
   def setAttributes(attributes: Attributes): TraversalBuilder = {
-    if (attributes ne Attributes.none) {
-      if (attributes.isAsync) this.makeIsland(GraphStageTag).internalSetAttributes(attributes)
-      else internalSetAttributes(attributes)
-    } else
-      this
+    if (attributes.isAsync) this.makeIsland(GraphStageTag).internalSetAttributes(attributes)
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/TraversalBuilder.scala]: Ah yes, you are right, this wanted to be an optimization to not do anything on Attributes.none, but this is handled elsewhere.

There is one worry now, that if you add a Dispatcher, then immediately set attributes to `none` then the async boundary remains.

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/TraversalBuilder.scala]: I don't mind btw, just saying. Alternative would be that the materializer scans every PushAttribute() and looks at async stuff. But that would null most of our performance I am afraid.

Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/TraversalBuilder.scala]: um, sounds like a weird thing to do, unless it's needed for fusing together a stage with IODispatcher as default together with a simple map stage. Shall we create a ticket about how to define async islands (might need some investigation of how it works and clarifications in documentation)?

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/TraversalBuilder.scala]: Better document than not, I guess.

Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/TraversalBuilder.scala]: ticket created, fill in more details if you like https://github.com/akka/akka/issues/22498

**Comment by drewhk** [comment]
LGTM

**Review by ktoso — APPROVED** [review]
LGTM, seems good

**Pull request merged**